examples/code-splitting-native-import-context/README.md

```diff
@@ -0,0 +1,394 @@
+# example.js
+
+This example illustrates how to leverage the `import()` syntax to create ContextModules which are separated into separate chunks for each module in the `./templates` folder.
+
+``` javascript
+
+async function getTemplate(templateName) {
+	try {
+		let template = await import(`./templates/${templateName}`);
+		console.log(template);
+	} catch(err) {
+		console.error("template error");
+		return new Error(err);
+	}
+}
+
+getTemplate("a");
+getTemplate("b");
+getTemplate("c");
+```
+
+# templates/
+
+* a.js
+* b.js
+* c.js
+
+All templates are of this pattern:
+
+``` javascript
+module.exports = function() {
+	return "This text was generated by template X";
+}
```

Comment:
@sokra as @Kovensky said, when Node ships ESM the ModuleNamespace Objects that are created for CommonJS may not have named properties for things. The discussions of the exact nature of CommonJS and creating a standard interop (standardized by Node, which is the default maintainer of CommonJS these days) is under the https://github.com/nodejs/node-eps repository and is under various agenda items in TC39. The standard interop will not land in ECMA262 just like how CommonJS is never going to land in ECMA262. However, there are JIT reasons why named imports from CommonJS may never land, and the future safe option is to *only* provide a `default` export that maps to `module.exports` for now. The existing ECMA262 ModuleNamespace Object type cannot fully map to CommonJS `module.exports` for now and a wrapper *must* be used to be spec compliant.